src/Disks/ObjectStorages/MetadataStorageFromPlainRewritableObjectStorage.cpp

```diff
@@ -22,34 +24,62 @@ MetadataStorageFromPlainObjectStorage::PathMap loadPathPrefixMap(const std::stri
 {
     MetadataStorageFromPlainObjectStorage::PathMap result;
 
-    RelativePathsWithMetadata files;
-    object_storage->listObjects(root, files, 0);
-    for (const auto & file : files)
+    ThreadPool & pool = getIOThreadPool().get();
+    ThreadPoolCallbackRunnerLocal<void> runner(pool, "PlainRWMetaLoad");
+    std::mutex mutex;
+
+    LoggerPtr log = getLogger("MetadataStorageFromPlainObjectStorage");
+
+    ReadSettings settings;
+    settings.enable_filesystem_cache = false;
+    settings.remote_fs_method = RemoteFSReadMethod::read;
+    settings.remote_fs_buffer_size = 1024;  /// These files are small.
+
+    LOG_DEBUG(log, "Loading metadata");
+    size_t num_files = 0;
+    for (auto iterator = object_storage->iterate(root, 0); iterator->isValid(); iterator->next())
```

Comment:
Speed-up by streaming listing of files.

Reply:
`AzureIteratorAsync` is broken: https://pastila.nl/?0064b711/17814a9da619ce4b04bc1bae89ef1d8d#DV1T4M6yV2abS8xTBlSjOQ==